provisioning/utils.go

```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetServerInternalIP(osclient osclientset.Interface) (string, error) {
+func GetServerInternalIP(osclient osclientset.Interface, config *metal3iov1alpha1.ProvisioningSpec, client kubernetes.Interface, targetNamespace string) (string, error) {
 	infra, err := osclient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	if err != nil {
 		err = fmt.Errorf("Cannot get the 'cluster' object from infrastructure API: %w", err)
 		return "", err
 	}
-
-	// NOTE(dtantsur): do we need to handle non-BareMetal platforms here?
-	if infra.Status.PlatformStatus.BareMetal == nil {
-		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.BareMetalPlatformType, osconfigv1.OpenStackPlatformType, osconfigv1.VSpherePlatformType:
+		// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
```

Comment:
The `BareMetal` part will be different for OpenStack and VSphere (yay for copy-pasting in Go)

Reply:
Well spotted and my bad... fixed now, thank you!